File: ios/Classes/core/DownloadCallbacks.swift
import Foundation

class DownloadCallbacks {
    var id: Int64 = 0

    func onIDReceived(_ id: Int64) {
        self.id = id
    }

    func onProgress(_ progress: Double) {}

    func onProgress(_ name: String, _ progress: Double) {}

    func onDownloadCompleted(_ path: String) {}

    func onDownloadError(_ errorMessage: String) {}
}

File: android/src/main/java/com/odehbros/flutter_file_downloader/core/DownloadCallbacks.java
package com.odehbros.flutter_file_downloader.core;

public class DownloadCallbacks {
    public long id;

    public void onIDReceived(final long id) {
        this.id = id;
    }

    public void onProgress(final double progress) {
    }

    public void onProgress(final String name, final double progress) {
    }

    public void onDownloadCompleted(final String path) {
    }

    public void onDownloadError(final String errorMessage) {
    }
}
